test/it/testbase.py

```diff
@@ -38,12 +40,41 @@ def set_up_and_clean_workspace(self):
         os.environ["WORKSPACE"] = workspace_dir
         os.chdir(workspace_dir)
 
-    def createContext(self): return SmContext(SmApplication(self.config_dir_override), None, False, False)
+    def setup_local_git(self):
+        workspace_dir = os.path.join(os.path.dirname(__file__), "workspace")
+        testapp_dir = os.path.join(os.path.dirname(__file__), "../testapps/basicplayapp")
+        os.makedirs(os.path.join(workspace_dir,  "git"))
+        shutil.copytree(testapp_dir, os.path.join(workspace_dir,  "git", "basicplayapp"))
+        os.chdir(os.path.join(workspace_dir, "git", "basicplayapp"))
+        command = "git init && git add . && git commit -m 'test'"
+        ps_command = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, universal_newlines=True)
+        ps_command.communicate()
+        os.chdir(workspace_dir)
```

Comment:
Wondered if it was possible for this to fail, somehow? Is the teardown guaranteed to always run?

Reply:
set_up_and_clean_workspace will delete the whole workspace folder when the test runs so shouldn't be an issue if teardown fails